pkg/secrets/strategy.go

```diff
@@ -130,18 +118,128 @@ func (s Source) MarshalYAML() (interface{}, error) {
 	return s.MarshalRaw(), nil
 }
 
-type StrategyList []SourceMap
+// SourceMapList is a list of mappings that can be access via index or the item name.
+type SourceMapList []SourceMap
 
-func (l StrategyList) Less(i, j int) bool {
+func (l SourceMapList) Less(i, j int) bool {
 	return l[i].Name < l[j].Name
 }
 
-func (l StrategyList) Swap(i, j int) {
+func (l SourceMapList) Swap(i, j int) {
 	tmp := l[i]
 	l[i] = l[j]
 	l[j] = tmp
 }
 
-func (l StrategyList) Len() int {
+func (l SourceMapList) Len() int {
 	return len(l)
 }
+
+// HasName determines if the specified name is defined in the set.
+func (l SourceMapList) HasName(name string) bool {
+	_, ok := l.GetByName(name)
+	return ok
+}
+
+// GetByName returns the resolution strategy for the specified name and a bool indicating if it was found.
+func (l SourceMapList) GetByName(name string) (SourceMap, bool) {
+	for _, item := range l {
+		if item.Name == name {
+			return item, true
+		}
+	}
+
+	return SourceMap{}, false
+}
+
+// GetResolvedValue returns the resolved value of the specified name and a bool indicating if it was found.
+// You must resolve the value before calling, it does to do resolution for you.
+func (l SourceMapList) GetResolvedValue(name string) (interface{}, bool) {
```

Comment:
could you use a type param like `T` here and then return a `T`? I'm not sure how deep you have to go into `SourceMap` to make that work, so maybe it's not worth it at the moment, not sure.

Reply:
yeah right now there's just one place in the code that uses this function, and casting the interface{} to a concrete type is easier than trying to get generics to work in this case.

But I'll keep it in mind if we start using it more. For the most part in Porter, we ignore the actual values of any parameters and just pass them around as interface{} to the bundle (the exception is the porter specific parameter that's inserted into every bundle built by porter, "porter-debug").